engine/engine.go

```diff
@@ -435,13 +435,23 @@ func (r *ruleEngine) runTaggingRules(ctx context.Context, infoRules []ruleMessag
 				mapRuleSets[ruleMessage.ruleSetName] = rs
 			}
 			if rs, ok := mapRuleSets[ruleMessage.ruleSetName]; ok {
-				violation.Effort = nil
 				violation.Category = nil
-				// we need to tie these incidents back to tags that created them
-				for tag := range tags {
-					violation.Labels = append(violation.Labels, fmt.Sprintf("tag=%s", tag))
+				if violation.Effort != nil && *violation.Effort > 0 {
+					// we need to tie these incidents back to tags that created them
+					for tag := range tags {
+						violation.Labels = append(violation.Labels, fmt.Sprintf("tag=%s", tag))
+					}
```

Comment:
Since this loop is repeated in both `if` and `else`, can it be taken out?

Reply:
I think that the only difference AFAICT is which list on the ruleset you are putting them on, is that correct?

Also, I assume that these rules will be run again, because I think we put them in both tagging rules and non tagging rules.

I think that we should:
1. Make sure that if a rule has a tag it only is added to the tagging rules.
2. Keep the for loop at the top level
3. make an if statement for the violation effort, and add it to violation's then use `continue` to run the item in the for loop
4. set the insight

I think that will make the code a little easier to read. If you want a example let me know!